### PR TITLE
add configmap names support for airflow  runtime version

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This PR implements a change to include config name support to mount volume and volume mounts automatically by providing a file name 

## Related Issues

- https://github.com/astronomer/issues/issues/6680

## Testing

QA should able to provide pre-created configmap with runtime and airflow versions list

## Merging

cherry-pick to release-0.36 , release-0.37

### CODE REVIEW

File: charts/astronomer/templates/_helpers.yaml [hunk 1]
```diff
@@ -275,12 +275,12 @@ s3:
 - name: houston-config-volume
   mountPath: /houston/config/local-production.yaml
   subPath: local-production.yaml
-{{ if .Values.houston.airflowReleasesConfig }}
+{{ if and .Values.houston.airflowReleasesConfig ( not .Values.houston.airflowReleasesconfigMapName ) }}
```
Comment (rishkarajgi) [inline charts/astronomer/templates/_helpers.yaml:278]:
```suggestion
{{ if and .Values.houston.airflowReleasesConfig ( not .Values.houston.airflowReleasesConfigMapName ) }}
```

Comment (rishkarajgi) [reply charts/astronomer/templates/_helpers.yaml:278]: please maintain camel case

File: charts/astronomer/values.yaml [hunk 2]
```diff
@@ -297,7 +297,9 @@ houston:
 
   # this option allows to override airflow releases config
   #
+  airflowReleasesconfigMapName: ~
```
Comment (rishkarajgi) [inline charts/astronomer/values.yaml:300]:
```suggestion
  airflowReleasesConfigMapName: ~
```

File: charts/astronomer/values.yaml [hunk 3]
```diff
@@ -328,7 +330,9 @@ houston:
     #   - 2.0.0-alpine3.10-onbuild
     #   channel: stable
 
+  runtimeReleasesconfigMapName: ~
```
Comment (rishkarajgi) [inline charts/astronomer/values.yaml:333]:
```suggestion
  runtimeReleasesConfigMapName: ~
```

**Review by rishkarajgi — CHANGES_REQUESTED** [review]
please add tests

File: charts/astronomer/values.yaml [hunk 4]
```diff
@@ -328,7 +330,9 @@ houston:
     #   - 2.0.0-alpine3.10-onbuild
     #   channel: stable
 
+  runtimeReleasesConfigMapName: ~
   RuntimeReleasesConfig:
```
Comment (danielhoherd) [inline charts/astronomer/values.yaml:334]: This should be lowercase R
```suggestion
  runtimeReleasesConfig:
```

**Review by Shubham-astro — APPROVED** [review]
Looks good

File: tests/chart_tests/test_houston_api_deployment.py [hunk 5]
```diff
@@ -218,3 +218,45 @@ def test_houston_configmap_with_RuntimeReleasesConfig_enabled(self, kube_version
             "mountPath": "/houston/astro_runtime_releases.json",
             "subPath": "astro_runtime_releases.json",
         } in c_by_name["houston"]["volumeMounts"]
+
+    def test_houston_configmap_with_airflow_and_runtime_configmap_name_enabled(self, kube_version):
+        """Validate that houston configmap and its embedded data with runtime and airflow configmap name defined
+        ."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_houston_api_deployment.py:224]:
```suggestion
        """Validate that houston configmap and its embedded data with runtime and airflow configmap name defined."""
```

File: charts/astronomer/templates/_helpers.yaml [hunk 6]
```diff
@@ -353,15 +353,15 @@ s3:
...
   configMap:
-    name: {{ .Values.houston.runtimeReleasesconfigMapName }}
+    name: {{ .Values.houston.runtimeReleasesConfigMapName }}
 {{- end }}
-{{- if .Values.houston.airflowReleasesconfigMapName  -}}
+{{- if .Values.houston.airflowReleasesConfigMapName  }}
 - name: certifiedversions
   configMap:
-    name: {{ .Values.houston.airflowReleasesconfigMapName  }}
+    name: {{ .Values.houston.airflowReleasesConfigMapName  }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/_helpers.yaml:364]:
```suggestion
    name: {{ .Values.houston.airflowReleasesConfigMapName }}
```

File: charts/astronomer/templates/_helpers.yaml [hunk 7]
```diff
@@ -353,15 +353,15 @@ s3:
...
 {{- end }}
-{{- if .Values.houston.runtimeReleasesconfigMapName -}}
+{{- if .Values.houston.runtimeReleasesConfigMapName }}
 - name: runtimeversions
   configMap:
-    name: {{ .Values.houston.runtimeReleasesconfigMapName }}
+    name: {{ .Values.houston.runtimeReleasesConfigMapName }}
 {{- end }}
-{{- if .Values.houston.airflowReleasesconfigMapName  -}}
+{{- if .Values.houston.airflowReleasesConfigMapName  }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/_helpers.yaml:361]:
```suggestion
{{- if .Values.houston.airflowReleasesConfigMapName }}
```

**Pull request merged**